Use Common Sanitizers

scbedd opened the pull request:
@JoshLove-msft party without concern.

scbedd committed:
target version of proxy with common sanitizers. update startup for the universal output

jsquire commented:
> @JoshLove-msft party without concern.

![image](https://github.com/Azure/azure-sdk-for-net/assets/913445/c6a13fd1-7573-4963-a73a-46d0c304f01b)

_(I actually have no idea what that comment is intended to mean.)_

scbedd committed:
target version of proxy with updated common sanitizers

scbedd commented:
> > @JoshLove-msft party without concern.
> 
> ![image](https://private-user-images.githubusercontent.com/913445/325979734-c6a13fd1-7573-4963-a73a-46d0c304f01b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQxNjgwMjEsIm5iZiI6MTcxNDE2NzcyMSwicGF0aCI6Ii85MTM0NDUvMzI1OTc5NzM0LWM2YTEzZmQxLTc1NzMtNDk2My1hNzNhLTQ2ZDBjMzA0ZjAxYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQyNlQyMTQyMDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jOWI1NjM4ZGRjNDlhODVhNzVlMzBmNWI0ZmNlNjdjYzQyN2Q3YzY3MTg2YjI3M2EwNWJmNzM2MjcwNzAyNTdlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.12GmJ9DyHXrvs4uzEEPkNTvy_53czoM9vgBLEibXlu8)
> 
> _(I actually have no idea what that comment is intended to mean.)_

Nothing will force push the branch and eliminate any changes he makes 👍

JoshLove-msft committed:
Remove id sanitizer

JoshLove-msft committed:
remove start arg

scbedd committed:
Update the proxy version to one that is incapable of outputting to
stderr

JoshLove-msft committed:
testing removal of sanitizers

scbedd committed:
target newest version of the proxy

azure-sdk commented:
**API change check**

API changes are not detected in this pull request.

scbedd committed:
newer test proxy with proper defaulting of default sanitizers for principalId and clientId

scbedd committed:
pin the correct version of the proxy

JoshLove-msft committed:
storage

JoshLove-msft committed:
use base64 key

JoshLove-msft committed:
fix communication

JoshLove-msft committed:
to/from

JoshLove-msft committed:
friendlyname

JoshLove-msft committed:
metrics advisor

JoshLove-msft committed:
remove redacted

JoshLove-msft committed:
copy source

scbedd committed:
newest proxy version. updating UserDelegationKey.SignedTid/SignedOid to empty guids. UserDelegation key VALUE to 0 in base64 (MA==), and reduction in activity of SAS URI sanitizer to just sig

JoshLove-msft committed:
storage sanitizers

JoshLove-msft committed:
cache guid string

JoshLove-msft committed:
cache empty guid string

JoshLove-msft committed:
mark attestation as liveonly

scbedd committed:
bump to newest version of the proxy. stderr is ENABLED on this proxy instance

scbedd committed:
default the proxy to boot without output to stderr

JoshLove-msft committed:
ARM tests

JoshLove-msft committed:
Merge branch 'target-common-sanitizers-proxy' of https://github.com/Azure/azure-sdk-for-net into target-common-sanitizers-proxy

JoshLove-msft committed:
Merge branch 'main' of https://github.com/Azure/azure-sdk-for-net into target-common-sanitizers-proxy

JoshLove-msft committed:
eh

JoshLove-msft committed:
sasUri and x-ms-request-id

JoshLove-msft committed:
df

JoshLove-msft committed:
more arm

JoshLove-msft committed:
Increase timeout due to more sanitizers being used

JoshLove-msft committed:
fix pattern

JoshLove-msft committed:
fix patterns

JoshLove-msft committed:
clean up

JoshLove-msft committed:
clean up and mark lab test liveonly

scbedd committed:
newest version of the proxy

scbedd committed:
use newest proxy with adjustments to token replacement

JoshLove-msft commented on eng/Packages.Data.props:
```diff
@@ -381,7 +381,7 @@
 
   <PropertyGroup>
-    <TestProxyVersion>1.0.0-dev.20240410.2</TestProxyVersion>
+    <TestProxyVersion>1.0.0-dev.20240506.4</TestProxyVersion>
```
```suggestion
    <TestProxyVersion>1.0.0-dev.20240506.5</TestProxyVersion>
```

JoshLove-msft committed:
Update eng/Packages.Data.props

scbedd committed:
ensure that all uses of regex globbing are +, add missing + to token regex

scbedd committed:
Merge branch 'target-common-sanitizers-proxy' of https://github.com/azure/azure-sdk-for-net into target-common-sanitizers-proxy

JoshLove-msft committed:
token

JoshLove-msft committed:
operation location

JoshLove-msft committed:
clean

JoshLove-msft committed:
df

heaths commented on sdk/keyvault/Azure.Security.KeyVault.Certificates/tests/CertificateClientLiveTests.cs:
```diff
@@ -772,6 +772,7 @@ public async Task VerifyUpdateCertificatePolicy()
         [RecordedTest]
         [TestCase("application/x-pkcs12")]
         [TestCase("application/x-pem-file")]
+        [LiveOnly(Reason = "Certificate data cannot be stored in test recording.")]
```
I'm not crazy about this messaging. It will need to be. Instead, a tracking issue should be opened and the reason can point to that as something we need to track and eventually re-enable these tests.

All key and secret material herein is created upon test execution and has no associated with accounts. This would tear a huge hole in testing coverage since Key Vault is all about secrets, keys, and certificates.

JoshLove-msft replied:
These don't need to be marked LiveOnly. I've removed the attribute.

JoshLove-msft committed:
remove LiveOnly from KV

JoshLove-msft committed:
remove additional unnecessary liveonly usages

JoshLove-msft committed:
workloads

JoshLove-msft committed:
mark recordedtest

JoshLove-msft committed:
liveonly

JoshLove-msft commented:
/azp run net - core - ci

JoshLove-msft committed:
Merge branch 'main' of https://github.com/Azure/azure-sdk-for-net into target-common-sanitizers-proxy

scbedd committed:
try a version of the proxy that has slightly adjusted local caching

JoshLove-msft committed:
Exclude search sanitizer

JoshLove-msft committed:
:qw
erge branch 'target-common-sanitizers-proxy' of https://github.com/Azure/azure-sdk-for-net into target-common-sanitizers-proxy

JoshLove-msft committed:
Merge branch 'main' of https://github.com/Azure/azure-sdk-for-net into target-common-sanitizers-proxy

JoshLove-msft committed:
fix pattern

JoshLove-msft committed:
ignore token sanitizer for search
